src/glmtools.jl

```diff
@@ -20,57 +180,157 @@ end
 function mueta{T<:Real}(::CloglogLink, η::T)
     max(eps(T), exp(η) * exp(-exp(η)))
 end
+function inverselink(::CloglogLink, η)
+    expη = exp(η)
+    μ = -expm1(-expη)
+    omμ = exp(-expη)   # the complement, 1 - μ
+    μ, max(realmin(μ), expη * omμ), max(realmin(μ), μ * omμ)
+end
 
 linkfun(::IdentityLink, μ) = μ
 linkinv(::IdentityLink, η) = η
-mueta(::IdentityLink, η) = 1
+mueta(::IdentityLink, η) = one(η)
+inverselink(::IdentityLink, η) = η, one(η), oftype(η, NaN)
 
 linkfun(::InverseLink, μ) = inv(μ)
 linkinv(::InverseLink, η) = inv(η)
 mueta(::InverseLink, η) = -inv(abs2(η))
+function inverselink(::InverseLink, η)
+    μ = inv(η)
+    μ, -abs2(μ), oftype(μ, NaN)
+end
+
+linkfun(::InverseSquareLink, μ) = inv(abs2(μ))
+linkinv(::InverseSquareLink, η) = inv(sqrt(η))
```

Comment:
Thanks for the careful review.

I used the expressions returned by `inverse.gaussian()` in R as the basis for my code.
```julia
julia> R"print(names(fam <- inverse.gaussian()))";
 [1] "family"     "link"       "linkfun"    "linkinv"    "variance"
 [6] "dev.resids" "aic"        "mu.eta"     "initialize" "validmu"
[11] "valideta"   "simulate"

julia> R"fam$linkfun"
RCall.RObject{RCall.ClosSxp}
function (mu)
1/mu^2
<environment: namespace:stats>


julia> R"fam$linkinv"
RCall.RObject{RCall.ClosSxp}
function (eta)
1/sqrt(eta)
<environment: namespace:stats>


julia> R"fam$mu.eta"
RCall.RObject{RCall.ClosSxp}
function (eta)
-1/(2 * eta^1.5)
<environment: namespace:stats>


julia> R"fam$variance"
RCall.RObject{RCall.ClosSxp}
function (mu)
mu^3
<bytecode: 0x560013a5a328>
<environment: 0x560013a5c4a8>
```
and it looks as if you are correct - the variance function differs from `dμ/dη` by a factor of `-1/2`.  I guess this must be the case because the variance function should be positive but this link and its inverse is monotone decreasing.

Of course the same is true for `InverseLink` which is the canonical link for the `Gamma` distribution.

This raises the interesting question of how the examples using these distributions and canonical links worked in the first place.

I'll see what I can find.